Source/WebGPU/WebGPU/Device.mm

```diff
@@ -285,6 +312,22 @@ static void captureFrame(id<MTLDevice> captureObject)
     return WGSL::vertexBufferIndexForBindGroup(groupIndex, maxIndex);
 }
 
+id<MTLBuffer> Device::newBufferWithBytes(const void *pointer, NSUInteger length, MTLResourceOptions options) const
+{
+    id<MTLBuffer> buffer = [m_device newBufferWithBytes:pointer length:length options:options];
+    ASSERT(buffer);
```

Comment:
Is it really right to ASSERT() success? Don't we need to handle the case where allocation failed?

Reply:
Yes we should handle that case. I'll file a bug to track that. The debug assert is to catch the failure when it happens as opposed to letting it go unnoticed.